src/output/tablestorage-deduplication.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable sort-keys */
+import { TableClient } from "@azure/data-tables";
+import * as EL from "@elastic/elasticsearch";
+import { defaultLog } from "@pagopa/winston-ts";
+import * as O from "fp-ts/Option";
+import * as TE from "fp-ts/TaskEither";
+import { constVoid, flow, pipe } from "fp-ts/lib/function";
+import { getTableDocument, upsertTableDocument } from "../utils/tableStorage";
+import {
+  IOutputDocument,
+  indexDocument,
+  updateIndexDocument
+} from "./elasticsearch/elasticsearch";
+export const tableStorageDeduplication = (
+  elasticClient: EL.Client,
+  tableClient: TableClient,
+  indexName: string
+) => (document: IOutputDocument): TE.TaskEither<Error, void> =>
+  pipe(
+    TE.Do,
+    defaultLog.taskEither.info(`tableStorageDeduplication => ${document}`),
+    defaultLog.taskEither.info(`creating table => ${indexName}`),
+    () =>
+      pipe(
+        () => tableClient.createTable,
```

Comment:
We should avoid to try creating table for each document. Maybe we can extract table initialization into a wrapper that return a function that handle document deduplication :)